### ISSUE TITLE
check_http -f nofollow -e 200 behavior has changed and is now failing

### ISSUE THREAD

**Issue by thomasvs** [nagios-plugins/nagios-plugins#377]
My servers on CentOS 7 recently received an updated package for

Name        : nagios-plugins-http
Version     : 2.2.1
Release     : 9git5c7eb5b9.el7

In this new release, behavior has changed for a combination of -f and -e.

# /usr/lib64/nagios/plugins/check_http -H (redacted) -I (redacted) -u /new/contact/ -e 200 -s "VRIJHEM 39B" --onredirect=follow -v
GET /new/contact/ HTTP/1.1
User-Agent: check_http/v2.2.1.git (nagios-plugins 2.2.1)
Connection: close
Host: arcreaties.be
Accept: */*

http://chewie:80/new/contact/ is 470 characters
STATUS: HTTP/1.1 301 Moved Permanently
**** HEADER ****
Date: Fri, 27 Apr 2018 01:58:55 GMT
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/5.4.16
X-Powered-By: PHP/5.4.16
Set-Cookie: wfvt_951087624=5ae283e033796; expires=Fri, 27-Apr-2018 02:28:56 GMT; path=/; httponly
X-Pingback: http://www.arcreaties.be/new/xmlrpc.php
Location: http://www.arcreaties.be/new/contact/
Access-Control-Allow-Origin: *
Content-Length: 0
Connection: close
Content-Type: text/html; charset=UTF-8
**** CONTENT ****

Status line output matched "200" - 
HTTP CRITICAL - Status line output matched "200" - # 

In other words, this check is now failing to output a full line, not outputting performance data, and claiming there's an error even though it thinks that the status code is the expected code.  My suspicion is that something in the code is short-circuiting causing it to skip the actual following.

Indeed, if I run the same check but removing -e 200, the redirect gets followed, and the eventual status output is:

HTTP OK: HTTP/1.1 200 OK - 28668 bytes in 6.403 second response time |time=6.402820s;;;0.000000 size=28668B;;;0

In the previous version I had installed, the first statement worked as expected.

**Comment by Madlohe** [comment]
#456 should fix at least part of this. 

@thomasvs, would you be able to compile from the "maint" branch and determine whether this is still an issue? I'll also try to reproduce this soon.

**Comment by jalbstmeijer** [comment]
Not sure if related, but between epel el6 rpm versions
nagios-plugins-2.2.1-4git.el6.src.rpm

and
nagios-plugins-2.2.2-2.20190926git1b8ad57.el6.src.rpm

similar behavior changed.

Before this followed the redirect and matched the sting "test"

check_http  -e 'HTTP/1.1 200,HTTP/1.1 3' -H xxx -u / -r 'test' --onredirect=follow

after the update, the redirect was not followed anymore, till changed to

check_http  -e 'HTTP/1.1 200' -H xxx -u / -r 'test' --onredirect=follow
